### PR TITLE
[fix](pipeline) Fix query hang up if limited rows is reached

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Follow-up for #35466.

We should assure closed tasks will not block other tasks.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: be/src/pipeline/exec/exchange_sink_buffer.cpp [hunk 1]
```diff
@@ -117,6 +116,12 @@ bool ExchangeSinkBuffer::can_write() const {
 }
 
 void ExchangeSinkBuffer::_set_ready_to_finish(bool all_done) {
```
Comment (yiguolei) [inline be/src/pipeline/exec/exchange_sink_buffer.cpp:118]: scan --> local exchange sink ----> queue()        local exchange source ---> Exchange Sink Buffer (dependency)  brpc(depdendency.set_ready())


PipelineTask ---> sink eos
---> clear blocking state

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: be/src/pipeline/exec/exchange_sink_operator.h [hunk 2]
```diff
@@ -100,6 +100,8 @@ class ExchangeSinkLocalState final : public PipelineXSinkLocalState<> {
     RuntimeProfile::Counter* compress_timer() { return _compress_timer; }
     RuntimeProfile::Counter* uncompressed_bytes_counter() { return _uncompressed_bytes_counter; }
     [[nodiscard]] bool transfer_large_data_by_brpc() const;
+    bool eos() const override { return _reach_limit.load(); }
```
Comment (yiguolei) [inline be/src/pipeline/exec/exchange_sink_operator.h:103]: rename to  is_finished

**Comment by Gabriel39** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by Gabriel39** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**